packages/graphback-codegen-schema/src/definitions/schemaDefinitions.ts

```diff
@@ -136,11 +136,13 @@ function getModelInputFields(schemaComposer: SchemaComposer<any>, modelType: Gra
   return inputFields;
 }
 
-export function buildFindOneFieldMap(modelType: GraphQLObjectType): GraphQLInputFieldMap {
+export function buildFindOneFieldMap(modelType: ModelDefinition, schemaComposer: SchemaComposer<any>): GraphQLInputFieldMap {
+  const { name, type } = modelType.primaryKey;
+
   return {
-    id: {
-      name: "id",
-      type: GraphQLNonNull(GraphQLID),
+    [name]: {
```

Comment:
Instead of having 
```graphql
type Query {
  getNote(id: ID): Note
}
```

all the time, now are able to detect the name and the type of the primary key.

so if the id is `_id: GraphbackObjectID`, the argument will be named the same e.g

```graphql
type Query {
  getNote(_id: GraphbackObjectID): Note
}
```

This required a change in the templates too.

Reply:
We can configure the `idField` in the current version of Offix, so this won't affect us. The same goes for the datatastore, which is just based of the schema.

Reply:
Good to know.

Reply:
I believe there was a strong reason for always having `id: ID` and mapping to the actual field at the resolver level. This is to ensure that the semantics of the schema to be static and comply with GraphQLCRUD.

If this is to be changed, it will need to discussed in GraphQLCRUD first IMO.

https://github.com/aerogear/graphback/pull/1769#issuecomment-665692229

Reply:
Yep. It should be id for get all the time.

Reply:
Happy to revert this change considering the historical context. But we'll need the typing information, otherwise this won't work e.g it needs to be `id: GraphbackObjectID`.

Reply:
You are right. If we do just ID and then manually build objectId in provider?

Reply:
That could work but:
- We'll be loosing typing info
- Validation provided by the `GraphbackObjectID` resolver or any any primary key scalar in use
- Possibly hitting the database for a primary key that's not valid
- Adds unneeded tracking ensuring that it works consistently (much like what we had before with the internal mapping, but less)
- Possible confusion with other input types e.g in delete/update mutation we have an input type with  `_id:  GraphbackObjectID` field, but for the getXXX, this is named `id`, with type `ID` leading to questions like, what's the type of the primary key

Reply:
> Possible confusion with other input types e.g in delete/update mutation we have an input type with _id: GraphbackObjectID field, but for the getXXX, this is named id, with type ID leading to questions like, what's the type of the primary key

This is the same with using custom IDs through the `@id` annotation. For example if the true model PK is `email: EmailAddress` the user still needs to pass the email address to the query as `id: 'hello@email.com'` or whatever. I do see the potential for confusion, but the docs do state this feature and provide an example with email as PK. Maybe we should also add an example with the PK as `_id: GraphbackObjectID`.

@wtrocki I understand the desire for a static schema to follow GraphQLCRUD correctly. Does it really matter what scalar is used though, so long as the field name remains static?

At the end of the day, libraries should be able to work together with the spec regardless of the scalar that is used?

```gql
getNote(id: ID)
getNote(id: EmailAdress)
getNote(id: String)
getNote(id: GraphbackObjectID)
getNote(id: Int)
```